### ISSUE TITLE
"Built-in units" may be confusing -- is it worth changing?

### ISSUE THREAD

**Issue by sbmlsecretary** [sbmlteam/sbml-specifications#125]
The spec uses the expression "the built-in units of X" when referring to the global units of volume, area, time, etc.  This seems a bit confusing.  Would the expression "model's global units of X" be more clear?  Is it worth changing this?

Reported by: mhucka

Original Ticket: "sbml/sbml-specifications//123":https://sourceforge.net/p/sbml/sbml-specifications//123

**Comment by sbmlsecretary** [comment]
Logged In: YES 
user\_id=1045203
Originator: NO

I do not think they are just global units. Any unit defined with a unitDefinition is global. The units in table 2 are really built-in. We do not need to define them in a unitDefinition. But more, we do not need to declare them. They are used by default anywhere relevant. 

Original comment by: lenov

**Comment by sbmlsecretary** [comment]
- **assigned_to**: nobody --> mhucka

Original comment by: mhucka

**Comment by sbmlsecretary** [comment]
Logged In: YES 
user\_id=641982
Originator: NO

I can see where both Mike and Nicolas are coming from :-\)

What about "model's built-in units of X" ??

Original comment by: sarahkeating

**Comment by sbmlsecretary** [comment]
Logged In: YES 
user\_id=862059
Originator: NO

Nicolas, is correct that all units are global in SBML. The naming build-in may not be the best but global is definitely incorrect. I was thinking about naming the 3 different unit groups when working on the schema for L3. 

This is what I came up with
1\) Base Units
2\) Predefined Units
3\) User Defined Units

The "Predefined Units" can be redefined/overridden by user. This naming scheme is for L2 as for L3 I would prefer to have everything explicit in the XML as that would not require to read the specs when interpreting the document.

To summarize "built-in" or "predefined" is OK with me "global" not. I would prefer predefined since build-in has a very static taste to me, i.e., I do not expect to be able to change it. However this is a minor issue and  I do not think its is worth changing the specs.





Original comment by: shoops

**Comment by sbmlsecretary** [comment]
Logged In: YES 
user\_id=1809275
Originator: NO

Actually there are more possible combinations. 

Definition of the units:

\- predefined base units \(like "metre"\). These could be called built-in \(but they are not in the spec\)
\- user defined \(e.g. "micromoles"\)

Scope of the units:

\- locally specified units \(the units is specified for a single object\)
\- modelwide specified units \("substance", "volume", etc. Called built-in in the specs \). These in turn can be default or redefined. 

This results in 4 combinations. 
If we really want to change the text, I wouldn´t call section 4.4.3 "Built-in units". I would suggest model-wide units instead \(if this is even proper english, i\`m not sure\). We could say that these have a built-in default value that can be changed. 

Original comment by: s_sahle

**Comment by sbmlsecretary** [comment]
Logged In: YES 
user\_id=641982
Originator: NO

I vote with Stefan for
1\) "Base units" - metre, ampere, ... 
2\) "Predefined units" - volume, area, etc...
3\) "User defined units" - units the user has defined :-\)

Original comment by: sarahkeating

**Comment by sbmlsecretary** [comment]
Logged In: YES 
user\_id=343670
Originator: YES

Sven makes some good points.  However, as far as naming, I am leaning towards Stefan's proposal, and using the terminology "base units" and "predefined units" \(and "user-defined units"\).  

Sven, would you be in agreement with that terminology too?



Original comment by: mhucka

**Comment by sbmlsecretary** [comment]
- **priority**: 5 --> 3

Original comment by: mhucka

**Comment by sbmlsecretary** [comment]
Logged In: YES 
user\_id=1809275
Originator: NO

Yes, that´s fine.

Original comment by: s_sahle

**Comment by sbmlsecretary** [comment]
- **status**: open --> pending

Original comment by: mhucka

**Comment by sbmlsecretary** [comment]
Logged In: YES 
user\_id=1045203
Originator: NO

I agree with the proposed change and that it should be done.

Original comment by: lenov

**Comment by sbmlsecretary** [comment]
Logged In: YES 
user\_id=1045203
Originator: NO

That is Stefan's terminology

Original comment by: lenov

**Comment by sbmlsecretary** [comment]
Logged In: YES 
user\_id=862059
Originator: NO

I agree with the proposed change and that it should be done.

Original comment by: shoops

**Comment by sbmlsecretary** [comment]
- **milestone**: 678078 --> Accepted:_Changes_without_conformance_implications
- **status**: pending --> open

Original comment by: mhucka

**Comment by sbmlsecretary** [comment]
I just counted and realized this issue has reached a 4/5 majority.  I'm therefore changing the group to "Accepted: changes without conformance implications".

Original comment by: mhucka

**Comment by sbmlsecretary** [comment]
- **status**: open --> pending

Original comment by: mhucka

**Comment by sbmlsecretary** [comment]
- **status**: pending --> closed

Original comment by: mhucka

**Comment by sbmlsecretary** [comment]
Done for L2v4r1.  \(Changes are SVN commit version 8194.\)

Original comment by: mhucka
